internal/backends/sqlite/collection.go

```diff
@@ -53,17 +55,29 @@ func (c *collection) Query(ctx context.Context, params *backends.QueryParams) (*
 	db := c.r.DatabaseGetExisting(ctx, c.dbName)
 	if db == nil {
 		return &backends.QueryResult{
-			Iter: newQueryIterator(ctx, nil),
+			Iter: newQueryIterator(ctx, nil, nil),
 		}, nil
 	}
 
 	meta := c.r.CollectionGet(ctx, c.dbName, c.name)
 	if meta == nil {
 		return &backends.QueryResult{
-			Iter: newQueryIterator(ctx, nil),
+			Iter: newQueryIterator(ctx, nil, nil),
 		}, nil
 	}
 
+	list, err := newDatabase(c.r, c.dbName).ListCollections(ctx, new(backends.ListCollectionsParams))
```

Comment:
I don't think we need to call `ListCollections` here to find out whether the collection is capped. The collection itself will contain the field `CappedSize`, if it's greater than zero, the collection is capped, otherwise it's not.

Reply:
Yes, `metadata`'s `Collection.Settings.CappedSize` can be used once https://github.com/FerretDB/FerretDB/pull/3614 is merged :+1: 

Same goes for `postgresql`, `metadata`'s `Collection.CappedSize` can be replaced here.